processor/src/main/java/org/mapstruct/ap/internal/model/CollectionAssignmentBuilder.java

```diff
@@ -0,0 +1,163 @@
+/**
+ *  Copyright 2012-2017 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
```

Comment:
Well we could theoretically move this to `org.mapstruct.ap.internal.model.property.mapping` package or maybe name it `PropertyMappingCollectionAssignmentBuilder`. What do you prefer?

Theoretically, I would refactor the entire assignment into it's own Builder that will invoke the needed builders for: `assignToCollection`, `assignToArray` and `assignToPlain`. `PropertyMapping` is getting big slowly :)